pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,61 @@
+package dns
+
+import (
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zones  []configv1.DNSZone
+		expect operatorv1.ConditionStatus
+	}{
+		{
+			name: "This is testing for a successful update of 1 record",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: operatorv1.ConditionFalse,
+		},
+		{
+			name: "check status",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones:  []configv1.DNSZone{{}},
```

Comment:
I tried the test case with no zones but this https://github.com/openshift/cluster-ingress-operator/blob/5b6cf73f3357fd7e2c6681acf1055c940710978d/pkg/operator/controller/dns/controller.go#L148  still gives a nil and the else condition  https://github.com/openshift/cluster-ingress-operator/blob/5b6cf73f3357fd7e2c6681acf1055c940710978d/pkg/operator/controller/dns/controller.go#L159   executes as the 	err := r.dnsProvider.Ensure(record, zone) returns nil . Do we need to define validation for record and zone for the Ensure method ?